ya_gpt_bot/services/impl/conversation_service.py

```diff
@@ -0,0 +1,101 @@
+"""Service to get and update conversations."""
+import datetime
+from typing import Callable
+
+from sqlalchemy import delete, func, insert, select
+from sqlalchemy import text as sa_text
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.ext.asyncio import AsyncEngine
+
+from ya_gpt_bot.db.entities.conversation import t_conversation
+
+func: Callable
+
+MIN_TS_SQL = sa_text(
+    """
+with agg as (
```

Comment:
`dedent` function may be used to fix unneeded indentation in such cases without moving all of the text to real start of the line